synapse/tests/test_cortex.py

```diff
@@ -3836,6 +3837,58 @@ async def test_cortex_mirror(self):
                     await core00.sync()
                     self.len(1, await core00.nodes('inet:ipv4=9.9.9.8'))
 
+    async def test_cortex_mirror_of_mirror(self):
+
+        with self.getTestDir() as dirn:
+
+            path00 = s_common.gendir(dirn, 'core00')
+            path01 = s_common.gendir(dirn, 'core01')
+            path02 = s_common.gendir(dirn, 'core02')
+
+            async with self.getTestCore(dirn=path00) as core00:
+                await core00.nodes('[ inet:ipv4=1.2.3.4 ]')
+
+            s_tools_backup.backup(path00, path01)
+            s_tools_backup.backup(path01, path02)
+
+            async with self.getTestCore(dirn=path00) as core00:
+
+                self.false(core00.conf.get('mirror'))
+
+                await core00.nodes('[ inet:ipv4=1.2.3.4 ]')
+                await core00.nodes('$lib.queue.add(hehe)')
+                q = 'trigger.add node:add --form inet:fqdn --query {$lib.queue.get(hehe).put($node.repr())}'
+                await core00.nodes(q)
+
+                url = core00.getLocalUrl()
+
+                core01conf = {'mirror': url}
+                async with await s_cortex.Cortex.anit(dirn=path01, conf=core01conf) as core01:
+                    url2 = core01.getLocalUrl()
+
+                    core02conf = {'mirror': url2}
+                    async with await s_cortex.Cortex.anit(dirn=path02, conf=core02conf) as core02:
+
+                        await core00.nodes('[ inet:fqdn=vertex.link ]')
+                        await core00.nodes('queue.add visi')
+
+                        await core01.sync()
+                        self.len(1, await core01.nodes('inet:fqdn=vertex.link'))
+
+                        await core02.sync()
+
+                        self.len(1, await core02.nodes('inet:fqdn=vertex.link'))
+
+                        logentrycount00 = await core00.nexsroot.index()
+                        logentrycount01 = await core01.nexsroot.index()
+                        logentrycount02 = await core02.nexsroot.index()
+
+                        self.eq(logentrycount00, logentrycount01)
+                        self.eq(logentrycount01, logentrycount02)
+
+                        self.eq(await core00.nexsroot.index(), await core01.nexsroot.index())
+                        self.eq(await core01.nexsroot.index(), await core02.nexsroot.index())
+
```

Comment:
Ideally, it would be good to show changes from the middle and end mirror being propagated to the others as well.

Reply:
Done